### ISSUE TITLE
Testing package main module resolves incorrect source map

### ISSUE THREAD

**Issue by Hoten** [terser-js/terser#305]
Set a debugger statement here:
```
diff --git a/test/mocha/sourcemaps.js b/test/mocha/sourcemaps.js
index 0cdf41cd..fcfff4ec 100644
--- a/test/mocha/sourcemaps.js
+++ b/test/mocha/sourcemaps.js
@@ -187,6 +187,7 @@ describe("sourcemaps", function() {
             assert.strictEqual(code, "var a=function(n){return n};");
         });
         it("Should work with max_line_len", function() {
+            debugger;
             var result = UglifyJS.minify(read("./test/input/issue-505/input.js"), {
                 compress: {
                     directives: false,
```

Run ndb:
```
ndb node node_modules/.bin/mocha test/mocha  --grep "Should work with max_line_len"
```

Step into `UglifyJS.minify`. Mappings will be incorrect.

This is b/c the mocha test requires `UglifyJS` with `../..`, which resolves to the root dir, which reads the package.json and uses `.main`, which is the minified bundle. If you set `.main` to `dist/bundle.js`, the mappings work.

ndb claims it is being mapped from `bundle.js` (see bottom right):
![image](https://user-images.githubusercontent.com/4071474/53753391-f7b58c00-3e65-11e9-86a6-2a7a8303f8db.png)

Unclear to me atm what's going on. Possible it's a bug in ndb, but the way UglifyJS is resolved in the tests seems suspect too.

**Comment by fabiosantoscode** [comment]
I've noticed this as well!

It happens with stack traces too. The lines are completely wrong.

I don't know why this is happening. It could be a Terser bug or a Rollup bug.

A way to isolate it would be to import `dist/bundle.js` directly instead of `dist/bundle.min.js`, to see if this still occurs. If it does not, then it's a Terser bug.

**Comment by Hoten** [comment]
I tried that just now, importing `dist/bundle.js` works but the minified file does not. However, I think the minified source file and map are fine. I uploaded them to https://sokra.github.io/source-map-visualization and it looks fine.

![image](https://user-images.githubusercontent.com/4071474/53773302-b55c7100-3e9e-11e9-8986-04ede9210f1a.png)

**Comment by Hoten** [comment]
a smaller repro, cuts out rollup:

save to `tmp.js`:
```
function test() {
	var aaaaa = 1;
	var bbbbb = 1;
	var ccccc = 1;
	var ddddd = 1;
	console.log(aaaaa + bbbbb + ccccc + ddddd);
}

debugger;
for (var i = 0; i < 10; i++) {
	test();
}
```

```
yarn prepare && npx cross-env TERSER_NO_BUNDLE=1 bin/uglifyjs tmp.js --source-map 'includeSources=true,url=tmp.min.js.map' -o dist/tmp.min.js
ndb dist/tmp.min.js # messed up mappings
```

--

I went back to v3.9.0, and the old way of bundling produced correct mappings for the repro in the OP. could be due to either the bundling process changes, or something related to how terser generates source maps. a clue :) Could bisect to see.

**Comment by Hoten** [comment]
works - 9fe68c6b
no work - 28c9fc31


```
git bisect start 28c9fc31 9fe68c6b
# add debugger statement as in OP repro ...
rm -rf dist && mkdir dist && npm install && (npm run prepublish || npm run prepare) && ndb node node_modules/.bin/mocha test/mocha --grep "Should work with max_line_len"
# see if mappings work in ndb

git checkout -- .
git bisect <good/bad>
```


Got me to commit e6e43800dc1de82ff58e9dd77aad349319e0aa03.


`git bisect log`:
```
# bad: [28c9fc31b396feecf88a0bbec519b8bfaa68ed51] 3.16.1
# good: [9fe68c6b2347286bfd1d47a85a76f1a92a47b769] 3.9.0
git bisect start '28c9fc31' '9fe68c6b'
# good: [5a25305ac121a4c45bdd0801bae33c7ffd925e4e] add build status
git bisect good 5a25305ac121a4c45bdd0801bae33c7ffd925e4e
# good: [5a25305ac121a4c45bdd0801bae33c7ffd925e4e] add build status
git bisect good 5a25305ac121a4c45bdd0801bae33c7ffd925e4e
# good: [757b0cfe4a1815e8fcbb7b7b81106a192055a3f7] compress dist library to speed up uglifyjs
git bisect good 757b0cfe4a1815e8fcbb7b7b81106a192055a3f7
# good: [4910a20837b24df60d67023fe3c51786f8f9a618] fix join_vars
git bisect good 4910a20837b24df60d67023fe3c51786f8f9a618
# good: [04305d820dde833f061b0cdaeadc8058e595624b] Return mochallel (#228)
git bisect good 04305d820dde833f061b0cdaeadc8058e595624b
# bad: [e6e43800dc1de82ff58e9dd77aad349319e0aa03] Use ES6 modules, build with rollup (#209)
git bisect bad e6e43800dc1de82ff58e9dd77aad349319e0aa03
# good: [aa574b6a02c6361a52565ffcbd328476f44099ec] Fix typo: protoype -> prototype (#232)
git bisect good aa574b6a02c6361a52565ffcbd328476f44099ec
# good: [81b535fddd66e047ac42c41cad85bb7a16a4c66b] Fix parsing of template literals with an escape (#231)
git bisect good 81b535fddd66e047ac42c41cad85bb7a16a4c66b
# good: [6f6b20194ebeac0c3592cfafec738565c4a6f9e9] Drop Node.js v4 (#207) (#226)
git bisect good 6f6b20194ebeac0c3592cfafec738565c4a6f9e9
# first bad commit: [e6e43800dc1de82ff58e9dd77aad349319e0aa03] Use ES6 modules, build with rollup (#209)
```

**Comment by Hoten** [comment]
Looking more at the `tmp.js` example, I'm more convinced this is not a bug in terser, but in ndb.

I uploaded `dist/tmp.min.js` and ran it through [sourcemapvalidator.io](https://sourcemaps.io/report/1551805425786_http%3A%2F%2Fmacho-degree.surge.sh%2Ftmp.min.js). Checks out.

I opened http://macho-degree.surge.sh/ in Chrome (stable and canary channels), and the mappings worked fine. Just not in ndb. So I'll file a bug there.
